packages/react/context-menu/src/ContextMenu.tsx

```diff
@@ -113,11 +115,14 @@ const ContextMenuTrigger = React.forwardRef<ContextMenuTriggerElement, ContextMe
         <MenuPrimitive.Anchor {...menuScope} virtualRef={virtualRef} />
         <Primitive.span
           data-state={context.open ? 'open' : 'closed'}
+          data-disabled={disabled ? '' : undefined}
           {...triggerProps}
           ref={forwardedRef}
           // prevent iOS context menu from appearing
           style={{ WebkitTouchCallout: 'none', ...props.style }}
           onContextMenu={composeEventHandlers(props.onContextMenu, (event) => {
+            // if trigger is disabled, enable the native Context Menu
+            if (disabled) return;
```

Comment:
Instead of those internal checks, could we conditionally pass the handlers like so:

```tsx
onContextMenu={disabled ? undefined : composeEventHandlers(props.onContextMenu, (event) => {
  // clearing the long press here because some platforms already support
  // long press to trigger a `contextmenu` event
  clearLongPress();
  handleOpen(event);
  event.preventDefault();
})}
```

We can do that for the 5 handlers to ensure there are no side-effects of disabling.

Reply:
This version would be incorrect - you'd have to only disable the "own" handler - without disabling the `props.onContextMenu`.

Reply:
oops, yep you're right, it needs to be:

```tsx
onContextMenu={disabled ? props.onContextMenu : …}
```

Reply:
On top of that - I would add this:
```js
useEffect(() => {
  if (disabled) clearLongPress()
}, [disabled])
```

Reply:
Yep, that's good for completeness too, like we do on unmount 👍

Reply:
https://github.com/paperstick/primitives/pull/2/files